lib/ansible/modules/network/ios/ios_logging.py

```diff
@@ -249,9 +271,20 @@ def parse_level(line, dest):
     return level
 
 
+def parse_trap(line, dest):
+    trap_group = ('emergencies', 'alerts', 'critical', 'errors',
+                  'warnings', 'notifications', 'informational', 'debugging')
+
+    if dest == 'host' and line in trap_group:
```

Comment:
According to the regex `match = re.search(r'logging (\S+)', line, re.M)`, the `dest` will be `trap` here.
Also if there is match found, `line` would be something like `logging trap warnings`. So checking `line in trap_group` doesn't make sense. The logic needs to be improved.